### PR TITLE
Add extension methods

### PR DESCRIPTION
**Pull request by pitriq**
Hey :) 

I've added some extension methods for these utilities.

### CODE REVIEW

**Comment by mateusfccp** [comment]
@dnfield Do we need something else or can we merge it?

File: lib/src/extensions.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  Path dash({
+    required CircularIntervalList<double> dashArray,
+    DashOffset? dashOffset,
+  }) {
+    return dashPath(this, dashArray: dashArray, dashOffset: dashOffset);
+  }
+}
+
+extension ParsePathExtension on String {
```
Comment (mateusfccp) [inline lib/src/extensions.dart:22]: I don't think this is something we want. It's somehow invasive and not even clear what it means. Even if we keep it, we should turn it into a method `toPath()` instead, as being a getter here is misleading.

File: lib/src/extensions.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  }) {
+    return trimPath(this, percentage, firstOnly: firstOnly, origin: origin);
+  }
+
+  Path dash({
+    required CircularIntervalList<double> dashArray,
+    DashOffset? dashOffset,
+  }) {
+    return dashPath(this, dashArray: dashArray, dashOffset: dashOffset);
+  }
```
Comment (mateusfccp) [inline lib/src/extensions.dart:19]:
```suggestion
  }) => dashPath(this, dashArray: dashArray, dashOffset: dashOffset);
```

File: lib/src/extensions.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import 'package:path_drawing/path_drawing.dart';
+
+extension PathDrawingExtension on Path {
+  Path trim(
+    double percentage, {
+    bool firstOnly = true,
+    PathTrimOrigin origin = PathTrimOrigin.begin,
+  }) {
+    return trimPath(this, percentage, firstOnly: firstOnly, origin: origin);
+  }
```
Comment (mateusfccp) [inline lib/src/extensions.dart:12]:
```suggestion
  }) => trimPath(this, percentage, firstOnly: firstOnly, origin: origin);
```

**Review by mateusfccp — COMMENTED** [review]
Some suggestions.

**Comment by dnfield** [comment]
I'd rather not add extensions to this library. IMHO they become pretty hard to reason about and document properly, and become kind of brittle.

**Pull request closed without merging**